cfme/tests/automate/custom_button/test_infra_objects.py

```diff
@@ -339,3 +380,57 @@ def test_custom_button_expression(appliance, request, setup_obj, button_group, e
             assert not custom_button_group.is_displayed
             setup_obj.add_tag(tag)
             assert custom_button_group.is_displayed
+
+
+@pytest.mark.uncollectif(lambda button_group: "VM_INSTANCE" not in button_group)
+def test_open_url(appliance, request, setup_obj, button_group, method):
+    """ Test Open url functionality of custom button.
+
+    Polarion:
+        assignee: ndhandre
+        initialEstimate: 1/2
+        caseimportance: high
+        testSteps:
+            prerequisites:
+                * Appliance with Infra provider
+
+            Steps:
+                * Create ruby method for url functionality
+                * Create custom button group with the Object type
+                * Create a custom button with open_url option and respective method
+                * Navigate to object Detail page
+                * Execute custom button
+                * Check new tab open or not with respective url
+    """
+
+    group, obj_type = button_group
+    button = group.buttons.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        open_url=True,
+        display_for="Single entity",
+        system="Request",
+        request=method.name,
+    )
+    request.addfinalizer(button.delete_if_exists)
+
+    view = navigate_to(setup_obj, "Details")
+    custom_button_group = Dropdown(view, group.hover)
+    assert custom_button_group.has_item(button.text)
+
+    # To-Do: Move windows handling functionality to browser
```

Comment:
We have the ConsoleMixin class to provide some browser interaction for selecting windows, which would be good to move into browser itself. 

https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/vm_console.py